test/FunctionalTests/Tests/DataServices/UnitTests/DesignT4UnitTests/CodeGenerationContextTest.cs

```diff
@@ -294,5 +294,74 @@ public void GetEdmxStringFromMetadataPathShouldThrowExceptionIfInputIsInvalidUri
                 Assert.AreEqual(ex.Message, "Only file, http, https schemes are supported for paths to metadata source locations.");
             }
         }
+
+        [TestMethod]
+        public void SetOneCustomHeaderCorrectlyShouldNotThrowException()
+        {
+            ODataT4CodeGenerator codegen = new ODataT4CodeGenerator();
+            string customHeaderString = @"Authorization:Bearer bearer-token";
+            codegen.ValidateAndSetCustomHttpHeadersFromString(customHeaderString);
+            Assert.IsNotNull(codegen.CustomHttpHeaders);
+            Assert.AreEqual(1, codegen.CustomHttpHeaders.Count);
+            Assert.AreEqual("Authorization:Bearer bearer-token", codegen.CustomHttpHeaders.FirstOrDefault());

```

Comment:
```suggestion
            Assert.AreEqual("Authorization:Bearer bearer-token", codegen.CustomHttpHeaders.First());

```
No reason why it should ever return default

Reply:
Done